### PR TITLE
Recover State Summary Correctly

### LINKED ISSUE #12208: Beacon api get header error: nil summary returned from the DB
Reported from a user and this was working in `3.2.2`
```
curl --silent localhost:3500/eth/v1/beacon/headers/6100326 | jq .
{
  "message": "Could not check if block is optimistic: nil summary returned from the DB",
  "code": 500
}
```

The state summary from DB will return nil if the state summary does not exist in the DB. Looking at the State Summary DB implementation, it will try the cache before it tries the DB. The state summary is saved next to the state and the block during regular sync and saved every block during the initial sync. The main question is how did end up in a condition where the block exists in the DB but the state summary does not exist in the DB

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Bug Fix

**What does this PR do? Why is it needed?**

In the event we have the referenced block in our database but not its summary, we fallback and correctly recover it.

**Which issues(s) does this PR fix?**

Fixes #12208 

**Other notes for review**

### CODE REVIEW

**Review by terencechain — APPROVED** [review]
Might worth a metric

**Comment by nisdas** [comment]
We don't have the metric in stategen, which is why I havent added in anything new. Basically just followed how we handled it in stategen

File: beacon-chain/blockchain/chain_info.go [hunk 1]
```diff
@@ -476,3 +482,18 @@ func (s *Service) Ancestor(ctx context.Context, root []byte, slot primitives.Slo
...
+			return nil, err
+		}
+		summary := &ethpb.StateSummary{Slot: b.Block().Slot(), Root: blockRoot[:]}
+		if err := s.cfg.BeaconDB.SaveStateSummary(ctx, summary); err != nil {
+			return nil, err
+		}
+		return summary, nil
+	}
+	return nil, errBlockDoesNotExist
+}
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/chain_info.go:499]: Want to add a test for this method?

**Review by prestonvanloon — APPROVED** [review]
LGTM

**Pull request merged**